plerkle/src/geyser_plugin_nft.rs

```diff
@@ -188,10 +201,18 @@ impl<T: 'static + Messenger + Default + Send + Sync> GeyserPlugin for Plerkle<'s
 
         let (sender, mut receiver) = mpsc::channel::<SerializedData>(32);
         self.sender = Some(sender);
-
+        let config: PluginConfig = Figment::new()
+            .join(Env::prefixed("PLUGIN_"))
+            .extract()
+            .map_err(|config_error|
+                GeyserPluginError::ConfigFileReadError {
+                    msg: format!("Could not read messenger config: {:?}", config_error)
+                }
+            )?;
         runtime.spawn(async move {
             // Create new Messenger connection.
-            if let Ok(mut messenger) = T::new().await {
+
+            if let Ok(mut messenger) = T::new(config.messenger_config).await {
```

Comment:
This is really cool, messenger stays generic because the config is generic, and its really simple to add the environment stuff in the docker yaml.